StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet.swift

```diff
@@ -13,6 +13,75 @@ import PassKit
 @_spi(STP) import StripePaymentsUI
 import UIKit
 
+protocol IntentAbstraction: PaymentMethodRequirementProvider {
+    /// Extracts all the recommended `PaymentMethodType`s from the given `intent`.
+    /// - Returns: An ordered list of all the `PaymentMethodType`s for this `intent`.
+    var recommendedPaymentSheetMethodTypes: [PaymentSheet.PaymentMethodType] { get }
+    
+    var recommendedPaymentMethodTypes: [STPPaymentMethodType] { get }
+    
+    var isSettingUp: Bool { get }
+    
+    var unactivatedPaymentMethodTypes: [STPPaymentMethodType] { get }
+    
+    var currency: String? { get }
+    
+    var amount: Int? { get }
+    
+    var supportsLink: Bool { get }
+    
+    var isPaymentIntent: Bool { get }
+    
+    var callToAction: ConfirmButton.CallToActionType { get }
```

Comment:
hm surprising, doesn't seem to be used

Reply:
Usage is here: https://github.com/stripe/stripe-ios/blob/master/StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentSheetViewController.swift#L348